artemis-commons/src/main/java/org/apache/activemq/artemis/utils/ActiveMQThreadPoolExecutor.java

```diff
@@ -46,32 +45,70 @@
    private static class ThreadPoolQueue extends LinkedBlockingQueue<Runnable> {
 
       private ActiveMQThreadPoolExecutor executor = null;
+      private int idleThreads = 0;
 
       public void setExecutor(ActiveMQThreadPoolExecutor executor) {
          this.executor = executor;
       }
 
       @Override
-      public boolean offer(Runnable runnable) {
+      public synchronized boolean offer(Runnable runnable) {
```

Comment:
I put the code into our product and tested the specific issue that a HA backup server is occasionally not taking over. So far, our product tests have not found new issues and I couldn't reproduce the HA backup failover issue. I did not run any performance tests.
BTW: I have rechecked the code and found a solution without any synchronization, just using an AtomicInteger. I'm currently testing my new changed and want to also discuss my changes with a collegue. I suggest to wait reviewing and testing my changes until I committed these changes later today.